extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/generator/BlocGenerator.kt

```diff
@@ -34,7 +39,7 @@ abstract class BlocGenerator(private val name: String,
     abstract fun fileName(): String
 
     fun generate(): String {
-        val substitutor = StrSubstitutor(templateValues)
+        val substitutor = StrSubstitutor(templateValues, "^{", "}", '\\')
```

Comment:
why is this substitution needed?

Reply:
in the template for the freezed code there's a '$' for the mixin class. The StrSubstitutor class uses '$' as prefix ${ and also as escape char. i wasn't able to make the substitution work for a template like `_$${bloc_camel_case}State` to `_$LoginState`. Therefore i changed the prefix to ^{ and the escape char to something else than '$'.

Reply:
Hmm is it not possible to escape the freezed $ like `\$`?

Reply:
Unfortunately not: the following template:
```
import 'package:freezed_annotation/freezed_annotation.dart';

part '${bloc_snake_case}_event.freezed.dart';

@freezed
class ${bloc_pascal_case}Event with _\$${bloc_pascal_case}Event {
  const factory ${bloc_pascal_case}Event.loadData() = ${bloc_pascal_case}LoadDataEvent;
}
```

with the default substitutor `val substitutor = StrSubstitutor(templateValues)`

results in:

```dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'checkin_event.freezed.dart';

@freezed
class CheckinEvent with _\${bloc_pascal_case}Event {
  const factory CheckinEvent.loadData() = CheckinLoadDataEvent;
}
```

If i add two '\' i get two in the generated code...

Reply:
weird...that's super annoying 😛